torch_geometric/loader/node_loader.py

```diff
@@ -143,8 +154,103 @@ def _get_iterator(self) -> Iterator:
         if self.filter_per_worker:
             return super()._get_iterator()
 
+        # if self.device.type == 'cpu' and not self.cpu_affinity_enabled:
+        # TODO: Add manual page for best CPU practices for PyG and switch on warning message
+        # link = ...
+        # Warning(f'Dataloader CPU affinity opt is not enabled, consider switching it on '
+        #             f'(see enable_cpu_affinity() or CPU best practices for PyG [{link}])')
         # Execute `filter_fn` in the main process:
         return DataLoaderIterator(super()._get_iterator(), self.filter_fn)
 
+    def __enter__(self):
+        return self
+
     def __repr__(self) -> str:
         return f'{self.__class__.__name__}()'
+
+    @contextmanager
+    def enable_cpu_affinity(self, loader_cores: List[int] = None):
```

Comment:
Is it possible to add a basic test for this?

Reply:
Added test `test_cpu_affinity_neighbor_loader()` in `test_neighbor_loader.py` 
https://github.com/JakubPietrakIntel/pytorch_geometric/blob/2ce828aa451bd2a4c9ad7c0deb8a7e0cee99437b/test/loader/test_neighbor_loader.py#L578-L599